### PR TITLE
Map TextureUtil

### PR DESCRIPTION
**Pull request by Sturmlilie**
Names were chosen to make the most sense to someone familiar with OpenGL.  
First time, please be gentle (◠︿◠✿)

### CODE REVIEW

File: mappings/net/minecraft/client/texture/TextureUtil.mapping [hunk 1]
```diff
@@ -1 +1,51 @@
...
+		ARG 4 height
+	METHOD method_24961 allocate (Lnet/minecraft/client/texture/NativeImage$GLFormat;IIII)V
+		ARG 0 internalFormat
+		ARG 1 id
+		ARG 2 levels
+		ARG 3 width
+		ARG 4 height
+	METHOD method_24962 (Ljava/io/InputStream;)Ljava/nio/ByteBuffer;
+		ARG 0 inputStream
+	METHOD method_24963 uploadImageBGRA (Ljava/nio/IntBuffer;II)V
```
Comment (liach) [inline mappings/net/minecraft/client/texture/TextureUtil.mapping]:
```suggestion
	METHOD method_24963 uploadImageBgra (Ljava/nio/IntBuffer;II)V
```

File: mappings/net/minecraft/client/texture/TextureUtil.mapping [hunk 2]
```diff
@@ -1 +1,51 @@
...
+	FIELD field_22547 LOGGER Lorg/apache/logging/log4j/Logger;
+	METHOD method_24956 generateId ()I
+	METHOD method_24957 deleteId (I)V
+		ARG 0 id
+	METHOD method_24958 allocate (III)V
+		ARG 0 id
+		ARG 1 width
+		ARG 2 height
+	METHOD method_24959 allocate (IIII)V
+		COMMENT
```
Comment (liach) [inline mappings/net/minecraft/client/texture/TextureUtil.mapping]:
```suggestion
```

**Comment by liach** [comment]
Also mind edit this pr so it merges into `1.16-pre2` instead?

**Comment by Sturmlilie** [comment]
@liach Thanks for the review, and sorry about the `Bgra`, I read the contribution guide afterwards but somehow didn't connect the dots :sweat_smile:

**Comment by Sturmlilie** [comment]
Actually don't merge yet please, I messed up the `Bgra` method.

**Comment by Sturmlilie** [comment]
The method was badly named, I interpreted the GL parameters wrong; `GL_BGRA` + `GL_UNSIGNED_INT_8_8_8_8_REV` = logical bit order is ARGB [1]. I completely removed the suffix because BGRA seemed like a curiosity; ARGB on the other hand is the standard hex format used everywhere else in the code too. Now it's fixed.

[1] https://stackoverflow.com/questions/7786187/opengl-texture-upload-unsigned-byte-vs-unsigned-int-8-8-8-8/7786405#7786405

**Review by liach — APPROVED** [review]

**Review by liach — APPROVED** [review]

**Review by YanisBft — APPROVED** [review]

**Review by Juuxel — APPROVED** [review]

**Pull request merged**